### PR TITLE
clear all internal variables starting with $, but not STATE() vars

### PR DESCRIPTION
**Pull request by arun3688**
### Purpose
Filter all internal variables starting with $, except STATE() vars to improve readability of modeldesiption.xml in FMI-2.0

### Changed Files:
SimCodeUtil.mo - check of all vars starting with $, at the same check if the variable is a STATE() var, as state vars should not be filtered.

### CODE REVIEW

File: testsuite/openmodelica/fmi/ModelExchange/2.0/fmi_attributes_06.mos [hunk 1]
```diff
@@ -70,68 +70,12 @@ readFile("fmi_attributes_06.xml"); getErrorString();
...
-//   <!-- Index of variable = \"9\" -->
-//   <ScalarVariable
 //     name=\"Fo\"
 //     valueReference=\"8\"
 //     >
 //     <Real/>
 //   </ScalarVariable>
-//   <!-- Index of variable = \"10\" -->
-//   <ScalarVariable
-//     name=\"x\"
```
Comment (lochel) [inline testsuite/openmodelica/fmi/ModelExchange/2.0/fmi_attributes_06.mos]: These states and corresponding derivatives shouldn't be suppressed.

File: testsuite/openmodelica/cppruntime/fmu/modelExchange/2.0/testDrumBoiler.mos [hunk 2]
```diff
@@ -157,148 +157,84 @@ val(evaporator_qm_S, 100);
 //   </ScalarVariable>
 //   <!-- Index of variable = \"7\" -->
 //   <ScalarVariable
-//     name=\"der(evaporator.T_D)\"
```
Comment (lochel) [inline testsuite/openmodelica/cppruntime/fmu/modelExchange/2.0/testDrumBoiler.mos]: These variables shouldn't be suppressed.

File: OMCompiler/Compiler/SimCode/SimCodeUtil.mo [hunk 3]
```diff
@@ -8113,6 +8113,19 @@ algorithm
...
+  output Boolean b;
+protected
+  String s;
+algorithm
+  b := match(cr)
+    case(DAE.CREF_QUAL(ident="$CLKPRE")) then false;
+    else false;
+  end match;
+end isClockPreviousFalse;
+
```
Comment (lochel) [inline OMCompiler/Compiler/SimCode/SimCodeUtil.mo]: Is the function used somewhere? Please remove it if not needed.

File: testsuite/openmodelica/fmi/ModelExchange/2.0/fmi_attributes_06.mos (hunk 1 above)
Comment (arun3688) [reply testsuite/openmodelica/fmi/ModelExchange/2.0/fmi_attributes_06.mos]: the variable x is a dummy state variable denoted by BackendDAE.DUMMY_STATE(), and its corresponding derivate der(x) and hence there is no derivative reference for the der(x), should dymmy states and derivative vars filtered or not ?

File: testsuite/openmodelica/cppruntime/fmu/modelExchange/2.0/testDrumBoiler.mos (hunk 2 above)
Comment (arun3688) [reply testsuite/openmodelica/cppruntime/fmu/modelExchange/2.0/testDrumBoiler.mos]: removed filtering of dummy states and dummy derivative vars

**Pull request merged**